pymobiledevice3/remote/remote_service_discovery.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import List, Tuple
+
+from pymobiledevice3.exceptions import NoDeviceConnectedError
+from pymobiledevice3.remote.bonjour import DEFAULT_BONJOUR_TIMEOUT, get_remoted_addresses
+from pymobiledevice3.remote.remotexpc import RemoteXPCConnection
+
+
+@dataclass
+class RSDDevice:
+    address: str
+    udid: str
+    product_type: str
+    os_version: str
+
+
+# from remoted ([RSDRemoteNCMDeviceDevice createPortListener])
+RSD_PORT = 58783
+
+
+class RemoteServiceDiscoveryService:
+    def __init__(self, address: Tuple[str, int]):
+        self.service = RemoteXPCConnection(address)
+        self.peer_info = None
+
+    def connect(self) -> None:
+        self.service.connect()
+        self.peer_info = self.service.receive_response()
+
+    def connect_to_service(self, name: str) -> 'RemoteXPCConnection':
```

Comment:
Why `'RemoteXPCConnection'` and not `RemoteXPCConnection`?

Reply:
fixed